[Maximilliano Utomo Quok] ip

tzaph opened the pull request:
# A.R.O.N.A.
> “Thanks for the hard work, Sensei.”

A.R.O.N.A. frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/tzaph/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [X] Managing tasks
- [x] Set up GUI (coming soon)
- [x] Add aliasing commands (coming soon)

---

If you are a Java programmer, you can check the repository out. Here's a snippet to find tasks with a specific keyword:

```Java
public void listTasksWithKeyword(String keyword) {
    int id = 1;
    ArrayList<String> linesToPrint = new ArrayList<>();
    linesToPrint.add("Here are the matching tasks in your list:");

    for (Task task : tasks) {
        if (task.hasKeyword(keyword)) {
            linesToPrint.add(id + "." + task);
        }
        id++;
    }

    if (id == 1) {
        linesToPrint.set(0, "Cannot find any tasks with this keyword");
    }

    String lines[] = new String[linesToPrint.size()];

    for (int i = 0; i < linesToPrint.size(); i++) {
        lines[i] = linesToPrint.get(i);
    }

    Ui.printLines(lines);
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

maxiutomoq committed:
level-0: Rename bot and add greet and exit feature

maxiutomoq committed:
level-0: rename file and modify gitignore

maxiutomoq committed:
level-1: echo command

maxiutomoq committed:
level-2: task list add and print feature

maxiutomoq committed:
level-3: done checkmark feature

maxiutomoq committed:
level-4: todo, deadline task, event task feature

maxiutomoq committed:
A-TextUiTesting: input test and expected output

maxiutomoq committed:
level-5: exception handling

maxiutomoq committed:
Level-6: delete task feature

maxiutomoq committed:
level-7: save task list feature

maxiutomoq committed:
level-8: enforce date format

maxiutomoq committed:
A-MoreOOP: Extract parts of code to separate classes

maxiutomoq committed:
Merge branch 'add-gradle-support' of https://github.com/tzaph/ip

maxiutomoq committed:
Add tests for TaskList, ToDo, and Parser

maxiutomoq committed:
Set up jar file

maxiutomoq committed:
Add more JavaDocs

maxiutomoq committed:
no message

maxiutomoq committed:
Revert "no message"

This reverts commit 46a26f3dfc77867f4cfe8a6ff51c8e1987423135.

maxiutomoq committed:
A-CodingStandard: fix some java code

maxiutomoq committed:
level-9: find task command

maxiutomoq committed:
Merge branch 'branch-A-CodingStandard'

maxiutomoq committed:
Fix conflict with branch level 9

larainezo commented on src/main/java/arona/task/Deadline.java:
```diff
@@ -0,0 +1,49 @@
+            throw new AronaInvalidDateException("");
+        }
+    }
+
```
For consistency, you may consider adding a JavaDoc for the toDataFormat() function :)

larainezo commented on src/main/java/arona/Arona.java:
```diff
@@ -0,0 +1,206 @@
+        tasks.addElements(task);
+
+        Ui.printLines("I've added this task, Sensei!",
+                "    " + task,
```
I like the themed messages!

larainezo commented on src/main/java/arona/Arona.java:
```diff
@@ -0,0 +1,206 @@
+        tasks.listTasksWithKeyword(keyword);
+    }
+
+    private boolean processCommand(CommandType commandType, String[] commandSplit) throws
```
I think "throws" should be brought down to the line below

tsulim replied:
Perhaps you should rename the method such that the name explains to the reader accurately and at a sufficient level of detail of what the method does?

larainezo reviewed (commented):
Great job! I think your code uses good OOP standards and your coding style is neat!

tsulim commented on src/main/java/arona/ui/Ui.java:
```diff
@@ -0,0 +1,32 @@
+     */
+    public static void byeUser() {
+        printLines("Thanks for the hard work, Sensei!");
+    }
```
Should this method name be written as a verb?

tsulim commented on src/main/java/arona/storage/Storage.java:
```diff
@@ -0,0 +1,103 @@
+     */
+    public Storage() {
+        try {
+            File dataFolder = new File("./data");
+            if (!dataFolder.exists()) {
+                dataFolder.mkdir();
+            }
+
+            File dataFile = new File("./data/data.txt");
+            if (!dataFile.exists()) {
+                dataFile.createNewFile();
```
I like how you have named `dataFolder` and `dataFile` similarly. 👍

tsulim commented on src/main/java/arona/storage/Storage.java:
```diff
@@ -0,0 +1,103 @@
+    public void readTasksFromData(TaskList taskList) throws IOException {
+        ArrayList<Task> tasks = new ArrayList<>();
+        File dataFile = new File("./data/data.txt");
+        Scanner istream = new Scanner(dataFile);
```
Perhaps a more intuitive variable name here?

tsulim commented on src/main/java/arona/storage/Storage.java:
```diff
@@ -0,0 +1,103 @@
+     */
+    public void writeTasks(TaskList taskList) {
+        try {
+            FileWriter ostream = new FileWriter("./data/data.txt");
```
Similar to the previous comment, perhaps a more intuitive variable name can be used here?

tsulim commented on src/main/java/arona/task/Deadline.java:
```diff
@@ -0,0 +1,49 @@
+    /**
+     * The due date of the task.
+     */
+    protected LocalDate deadline;
```
Not very sure about this but perhaps you should avoid using case to distinguish the names? 😅

tsulim commented on src/main/java/arona/task/Event.java:
```diff
@@ -0,0 +1,57 @@
+ * @author Maximilliano Utomo
+ */
+public class Event extends Task {
+    /**
+     * The start time of the task.
+     */
+    LocalDate startTime;
+    /**
+     * The end time of the task.
+     */
+    LocalDate endTime;
```
Should this be `startDate` instead of `startTime`? Similarly, `endDate` instead of `endTime`?

tsulim reviewed (commented):
Hello there! 👋 

Great job! Most codes are in line with the [Coding Quality: Naming](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/topics.html#w4-6-code-quality-naming) that we learnt so far. Overall, I found your code easy to read apart from some portions where some minor amendment might be required. 🛠️ 

Excellent job at keeping up with the weekly iP tasks. 💯
Keep up the good work and I wish you all the best! 💪✨

maxiutomoq committed:
Level-10: Add GUI

maxiutomoq committed:
Update build.gradle

maxiutomoq committed:
Add assertions to code

maxiutomoq committed:
Improve code quality

tzaph committed:
Merge pull request #2 from tzaph/branch-A-Assertions

Add assertions to code

tzaph committed:
Merge branch 'master' into branch-A-CodeQuality

tzaph committed:
Merge pull request #3 from tzaph/branch-A-CodeQuality

Improve code quality

maxiutomoq committed:
Add short alias for commands and delete text-ui-test

maxiutomoq committed:
Improve GUI

The profile pictures are now clipped in a circle.

The window width is reduced for compactability

maxiutomoq committed:
Add user guide and finalize GUI

maxiutomoq committed:
Revert "Add user guide and finalize GUI"

This reverts commit 414439c97ea7621c724b8d632f9e72c9d0c8aab4.

maxiutomoq committed:
Add user guide and finalize GUI

Make the GUI more compact and fix small issues in the format.

Add a documentation user guide for users to understand how to use the
chatbot.

maxiutomoq committed:
Fix shadowJar build to generate the final jar

Also, add gui application title :D

maxiutomoq committed:
Fix jar build to support linux and mac
